pkg/daemon/ceph/osd/daemon.go

```diff
@@ -281,19 +281,27 @@ func Provision(context *clusterd.Context, agent *OsdAgent, crushLocation string)
 
 func getAvailableDevices(context *clusterd.Context, desiredDevices []DesiredDevice, metadataDevice string, pvcBacked bool) (*DeviceOsdMapping, error) {
 
+	var partCount int
+	var ownPartitions bool
+	var err error
+	var fs string
 	available := &DeviceOsdMapping{Entries: map[string]*DeviceOsdIDEntry{}}
 	for _, device := range context.Devices {
-		if device.Type == sys.PartType {
-			continue
-		}
-		partCount, ownPartitions, fs, err := sys.CheckIfDeviceAvailable(context.Executor, device.Name, pvcBacked)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to get device %q info", device.Name)
+		if device.Type != sys.PartType {
+			partCount, ownPartitions, fs, err = sys.CheckIfDeviceAvailable(context.Executor, device.Name, pvcBacked)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get device %q info", device.Name)
+			}
+
+			if fs != "" || !ownPartitions {
```

Comment:
Do we need to check for `fs != ""` here? The check for `device.Filesystem != ""` should also include this on line 303, right? It may simplify if this check is just for the partitions not being owned.

Reply:
Yes `fs != ""` can be removed I think. I just re-used what we already present. I don't think it hurts though.